rise-data.html

```diff
@@ -105,6 +111,83 @@
 
         // log usage
         this.$.logger.log(BQ_TABLE_NAME, params);
+      },
+
+      /**
+       * Cache the data
+       *
+       * @param {String} key The key to identify the item of data
+       * @param {Object} data The data to cache
+       * @param {Boolean} ts Optionally include a timestamp to cache
+       */
+      saveItem: function(key, data, ts) {
+        var cacheObj = {};
+
+        if (this._pingReceived && key && data) {
+          if (!this._isCacheRunning) {
+            if (supportsLocalStorage()) {
+              cacheObj.data = data;
+
+              if (ts) {
+                cacheObj.timestamp = moment().format();
+              }
+
+              try {
+                localStorage.setItem(key, JSON.stringify(cacheObj));
+              } catch(e) {
+                console.warn(e.message);
```

Comment:
@donnapep A typical scenario of if failing will be because the localStorage exceeds the 5mb capacity and logging the error message is helpful in this situation for other consumers of our components. We could use logger but that would only be for our benefit.